tests/foreman/maintain/test_upgrade.py

```diff
@@ -83,6 +95,69 @@ def test_positive_repositories_validate(sat_maintain):
     assert skip_message in result.stdout
 
 
+@pytest.mark.no_containers
+@pytest.mark.parametrize(
+    'custom_host',
+    [
+        {'deploy_rhel_version': '8', 'deploy_flavor': 'satqe-ssd.disk.xxxl'},
+        {'deploy_rhel_version': '8', 'deploy_flavor': 'satqe-ssd.standard.std'},
+    ],
+    ids=['default', 'medium'],
+    indirect=True,
+)
+def test_negative_pre_upgrade_tuning_profile_check(request, custom_host):
+    """Negative test that verifies a satellite with less than
+    tuning profile hardware requirements fails on pre-upgrade check.
+
+    :id: 240bb97e-7353-4397-8d0e-228c3593c8cc
+
+    :steps:
+        1. Check out a RHEL instance with less than tuning requirements
+        2. Install Satellite
+        3. Run pre-upgrade check
+
+    :expectedresults: Pre-upgrade check fails.
+    """
+    profile = request.node.callspec.id
+    sat_version = ".".join(settings.server.version.release.split('.')[0:2])
+    # Register to CDN for RHEL8 repos, download and enable last y stream's ohsnap repos,
+    # and enable the satellite module and install it on the host
+    custom_host.register_to_cdn()
+    last_y_stream = last_y_stream_version(settings.server.version.release)
+    custom_host.download_repofile(product='satellite', release=last_y_stream)
+    custom_host.execute('dnf -y module enable satellite:el8 && dnf -y install satellite')
+    # If testing <= 6.13 we can run installer with --disable-system-checks,
+    # otherwise we need to set the --tuning to development and then change the tuning
+    # profile manually to get around the installer check
+    if float(sat_version) <= 6.13:
+        custom_host.execute(
+            f'satellite-installer --scenario satellite --disable-system-checks --tuning {profile}',
+            timeout='30m',
+        )
+    else:
+        custom_host.execute(
+            'satellite-installer --scenario satellite --tuning development', timeout='30m'
+        )
+        custom_host.execute(
+            f'sed -i "s/tuning: development/tuning: {profile}/g" '
+            '/etc/foreman-installer/scenarios.d/satellite.yaml'
+        )
+    # Get current Satellite version's repofile and update foreman maintain
+    custom_host.download_repofile(product='satellite', release=sat_version)
+    custom_host.execute('dnf upgrade -y satellite-maintain rubygem-foreman_maintain')
```

Comment:
run `satellite-maintain upgrade list-versions` twice instead to trigger an update of `rubygem-foreman_maintain`

Reply:
I felt like it was more reliable to just directly update the package, but if you think having it update itself is better then I can switch.

Reply:
The benefit of Lukas's suggestion is you end up following what we tell users to do which helps double check that workflow (https://access.redhat.com/documentation/en-us/red_hat_satellite/6.12/html/upgrading_and_updating_red_hat_satellite/upgrading_satellite#upgrading_a_connected_satellite_server)

Reply:
@ehelms yep that was what I was assuming a user would do. There's really no downside to updating this way. My concern was that reviewers would find it odd to use that command to update.